src/python/grpcio/grpc/_cython/_cygrpc/call.pyx.pxi

```diff
@@ -106,6 +107,8 @@ cdef class Call:
   def __dealloc__(self):
     if self.c_call != NULL:
       grpc_call_destroy(self.c_call)
+    grpc_shutdown()
```

Comment:
I would be convinced that this approach solves the problem if you ran the helloworld example through gdb and saw ```grpc_cq_global_shutdown()``` getting called.

I attempted to do it locally but I think there are some issues with the run_tests.py.